static/js/timerender.js

```diff
@@ -40,6 +40,46 @@ exports.render_now = function (time) {
     return [time.toString("MMM\xa0dd"), false];
 };
 
+exports.last_seen_user_info = function (user_id) {
+    if (presence.get_status(user_id) === "active") {
+        return i18n.t("Active now");
+    }
+
+    if (people.get_person_from_user_id(user_id).is_bot) {
+        return i18n.t("Bot");
+    }
+
+    var last_active_date = presence.last_active_date(user_id);
+    var start_of_today = set_to_start_of_day(new XDate());
```

Comment:
We should have `timerender.js` deal just with time-based logic to avoid circular dependencies.  So I would move the first 7 or 8 lines of this code to `popovers.js`, and then have this function just handle all the variations of "Last seen a minute ago," "Last seen yesterday," etc.  And then it should be fairly easy to unit test, and we should have tests for this.